components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolver.java

```diff
@@ -49,6 +49,12 @@ public SCIMUserStoreException resolve(UserStoreException e) {
                 .ErrorMessages.ERROR_CODE_INVALID_DOMAIN_NAME.getCode().equals(((UserStoreClientException) e)
                         .getErrorCode())) {
             return new SCIMUserStoreException("Unable to proceed. Invalid domain name.", HttpStatus.SC_BAD_REQUEST);
+        } else if (e instanceof org.wso2.carbon.user.core.UserStoreClientException) {
+            String description = e.getMessage();
+            if (StringUtils.isBlank(description)) {
+                description = "Error occurred while performing the operation";
```

Comment:
"Error occurred while performing the operation" implies a server error. Shall we change the message to imply this is a client error? e.g. Invalid Request

Reply:
Fixed with 1ba5ff79d15a712a8a788bd071ed74c580c49be7